VMware: Adding IP-Specific properties to ESXi host firewall config

alongchamps opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Implements 51240 so that admins can configure firewall rules with IP and Network information.

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
vmware_host_firewall_manager

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

alongchamps committed:
Adds IP-Specific properties to firewall config

Implements ansible ticket 51240 so that admins can configure firewall rules on a per-rule level with IP-specific information.

alongchamps committed:
Adds IP-Specific properties to firewall config

Implements ansible ticket 51240 so that admins can configure firewall rules on a per-rule level with IP-specific information.

alongchamps committed:
Merge branch 'devel' of https://github.com/alongchamps/ansible-vmware-host-firewall-manager into devel

alongchamps committed:
Update main.yml

Changes hostvars ESXi list to use esxi1, esxi2

alongchamps committed:
Fixed indentation

Fixed indentation on main.yaml for tests and also removed unicode usage.

alongchamps committed:
Fix vmware_host_firewall_manager.py indentation

Fix vmware_host_firewall_manager.py indentation in RETURN block

goneri commented on lib/ansible/modules/cloud/vmware/vmware_host_firewall_manager.py:
```diff
@@ -185,11 +263,87 @@ def ensure(self):
+                playbook_allowed_networks = set(rule_config[0].get('ip_network', ''))
+
+                # compare allowed all IPs boolean to playbook
+                if rule_allows_all != playbook_allows_all:
```
You can write that this way:
`allow_all_ips_different = bool(rule_allows_all != playbook_allows_all)`

alongchamps replied:
Fixed on next commit to be one line

goneri commented on lib/ansible/modules/cloud/vmware/vmware_host_firewall_manager.py:
```diff
@@ -185,11 +263,87 @@ def ensure(self):
+                else:
+                    allowed_all_ips_different = False
+
+                # compare IPs in list to desired IPs
```
The same

alongchamps replied:
Fixed on next commit to be one line

goneri commented on lib/ansible/modules/cloud/vmware/vmware_host_firewall_manager.py:
```diff
@@ -185,11 +263,87 @@ def ensure(self):
+                else:
+                    ip_list_different = False
+
+                # compare networks in list to desired networks
```
The same

alongchamps replied:
Fixed on next commit to be one line

goneri commented on lib/ansible/modules/cloud/vmware/vmware_host_firewall_manager.py:
```diff
@@ -185,11 +263,87 @@ def ensure(self):
+                                    tmp_ip_network_spec.prefixLength = int(allowed_network.split("/")[1])
+                                    firewall_spec.allowedHosts.ipNetwork.append(tmp_ip_network_spec)
+
+                            print("applying")
```
Debug print()

alongchamps replied:
Removed on next commit.

goneri commented on lib/ansible/modules/cloud/vmware/vmware_host_firewall_manager.py:
```diff
@@ -138,6 +185,12 @@ def gather_rule_set(self):
                     temp_rule_dict['enabled'] = rule_set_obj.enabled
+                    allowed_host = rule_set_obj.allowedHosts
+                    rule_allow_host = dict()
+                    rule_allow_host['ip_address'] = [ip for ip in allowed_host.ipAddress]
```
I'm not sure why you loop on allowed_host.ipAddress if it's already a list.

alongchamps replied:
Changed to remove list logic and just assign the property.

goneri commented on lib/ansible/modules/cloud/vmware/vmware_host_firewall_manager.py:
```diff
@@ -167,6 +222,28 @@ def ensure(self):
+                    for ip_addr in rule_config[0]['ip_address']:
+                        try:
+                            ip = ipaddress.ip_address(ip_addr)
+                            print("ip is %s" % ip)
```
debug print()

alongchamps replied:
Removed on next commit.

goneri reviewed (changes requested):
Hi @alongchamps, thanks for the contribution. It took the freedom to do a quick review even in the PR has the Draft flag. For the record, you branch needs a rebase.

alongchamps committed:
Cleaning up

Cleaned up some leftover debugging statements and shortened some boolean decisions. Also hopefully fixed the example return object and fleshed out the returns examples.

alongchamps commented:
@goneri I did do a merge with the changes on the integration test file, main.yml. I think that would get rid of the need to rebase and make it clean?

goneri commented:
Yes, both are find, I personally prefer to rebase since it keeps the patches of my branch on top of the others.
